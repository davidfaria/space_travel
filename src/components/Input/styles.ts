import styled from 'styled-components'

export const Wrapper = styled.div`
  background: var(--background-input);
  border-radius: 4px;
  padding: 1.2rem;
  width: 100%;
  color: #999;

  display: flex;
  align-items: center;

  & + div {
    margin-top: 8px;
  }

  input {
    flex: 1;
    background: transparent;
    box-shadow: 0 0 0 0;
    border: 0 none;
    outline: 0;
    color: var(--text);
    &::placeholder {
      color: var(--placeholder);
    }
    &::focus {
      outline-offset: 0;
    }
  }

  svg {
    margin-right: 16px;
  }
`
