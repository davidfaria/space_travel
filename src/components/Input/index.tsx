import React, { InputHTMLAttributes } from 'react'
import { IconBaseProps } from 'react-icons'
import * as S from './styles'

export type InputProps = {
  name: string
  icon: React.ComponentType<IconBaseProps>
} & InputHTMLAttributes<HTMLInputElement>

const Input = ({ name, icon: Icon, ...props }: InputProps) => {
  const inputRef = React.useRef<HTMLInputElement>(null)

  return (
    <S.Wrapper>
      {Icon && <Icon size={20} />}
      <input data-testid={name} name={name} ref={inputRef} {...props} />
    </S.Wrapper>
  )
}

export default Input
