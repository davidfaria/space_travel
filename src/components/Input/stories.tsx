import { Story, Meta } from '@storybook/react/types-6-0'
import { MdSearch } from 'react-icons/md'
import Input from '.'

export default {
  title: 'Input',
  component: Input
} as Meta

export const Default: Story = () => (
  <Input
    type="search"
    name="search"
    icon={MdSearch}
    placeholder="Digite a distÃ¢ncia: ex: 1000000"
  />
)
