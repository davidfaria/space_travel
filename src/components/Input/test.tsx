import { render, screen, fireEvent } from '@testing-library/react'
import { MdSearch } from 'react-icons/md'

import Input from '.'

describe('<Input />', () => {
  it('should render the value when typed', async () => {
    const { container } = render(
      <Input
        type="search"
        name="search"
        icon={MdSearch}
        placeholder="Digite a distância: ex: 1000000"
      />
    )

    const input = screen.getByRole('searchbox')

    expect(
      screen.getByPlaceholderText(/Digite a distância: ex: 1000000/i)
    ).toBeInTheDocument()

    expect(input).toBeInTheDocument()

    fireEvent.change(input, {
      target: { value: '1000000' }
    })

    expect(input).toHaveValue('1000000')

    expect(container.firstChild).toMatchSnapshot()
  })
})
