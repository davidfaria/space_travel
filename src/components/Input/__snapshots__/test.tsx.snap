// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render the value when typed 1`] = `
.c0 {
  background: var(--background-input);
  border-radius: 4px;
  padding: 1.2rem;
  width: 100%;
  color: #999;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 + div {
  margin-top: 8px;
}

.c0 input {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  background: transparent;
  box-shadow: 0 0 0 0;
  border: 0 none;
  outline: 0;
  color: var(--text);
}

.c0 input::-webkit-input-placeholder {
  color: var(--placeholder);
}

.c0 input::-moz-placeholder {
  color: var(--placeholder);
}

.c0 input:-ms-input-placeholder {
  color: var(--placeholder);
}

.c0 input::placeholder {
  color: var(--placeholder);
}

.c0 input::focus {
  outline-offset: 0;
}

.c0 svg {
  margin-right: 16px;
}

<div
  class="c0"
>
  <svg
    fill="currentColor"
    height="20"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 24 24"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
    />
  </svg>
  <input
    data-testid="search"
    name="search"
    placeholder="Digite a distÃ¢ncia: ex: 1000000"
    type="search"
  />
</div>
`
